### ISSUE TITLE
Client stub should not take Metadata headers parameter

### ISSUE THREAD

**Issue by xiaodongw** [grpc/grpc-kotlin#47]
The current generated stub takes headers as parameter
```
suspend fun sayHello(request: HelloRequest, headers: Metadata = Metadata()): HelloReply
```

To my understanding, the headers is more for facilities like interceptors. For example, tracing, client side interceptor puts trace ID into headers, then server side interceptor consumes the trace ID. People may misuse it to carry business logic into it by exposing it on stub.

**Comment by lowasser** [comment]
No real objection.

**Comment by chalin** [comment]
Closed by #49 @lowasser?
